app/src/main/java/org/systers/mentorship/viewmodels/SignUpViewModel.kt

```diff
@@ -26,49 +22,75 @@ class SignUpViewModel : ViewModel() {
+            message = response.message
+            successful.value = true
+        } catch (throwable: Exception) {
+            when (throwable) {
+                is IOException -> {
+                    message = appContext.getString(R.string.error_please_check_internet)
+                }
+                is TimeoutException -> {
+                    message = appContext.getString(R.string.error_request_timed_out)
+                }
+                is HttpException -> {
+                    message = CommonUtils.getErrorResponse(throwable).message
+                }
+                else -> {
+                    message = appContext.getString(R.string.error_something_went_wrong)
+                    Log.e(TAG, throwable.localizedMessage)
+                }
+            }
+            successful.value = false
+        }
+    }
+
     /**
-     * Will be used to run the register method of the AuthService
+     * Used used to run the register method of the AuthService
      * @param register a registration request object containing the a user's registration fields
      */
-    @SuppressLint("CheckResult")
-    fun register(@NonNull register: Register) {
-        authDataManager.register(register)
-                .subscribeOn(Schedulers.newThread())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribeWith(object : DisposableObserver<CustomResponse>() {
-                    override fun onNext(customResponse: CustomResponse) {
-                        message = customResponse.message
-                        successful.value = true
-                    }
-
-                    override fun onError(throwable: Throwable) {
-                        when (throwable) {
-                            is IOException -> {
-                                message = MentorshipApplication.getContext()
-                                        .getString(R.string.error_please_check_internet)
-                            }
-                            is TimeoutException -> {
-                                message = MentorshipApplication.getContext()
-                                        .getString(R.string.error_request_timed_out)
-                            }
-                            is HttpException -> {
-                                message = CommonUtils.getErrorResponse(throwable).message
-                            }
-                            else -> {
-                                message = MentorshipApplication.getContext()
-                                        .getString(R.string.error_something_went_wrong)
-                                Log.e(TAG, throwable.localizedMessage)
-                            }
-                        }
-                        successful.value = false
-                    }
-
-                    override fun onComplete() {
-                    }
-                })
-    }
+//    @SuppressLint("CheckResult")
```

Comment:
If this is redundant code then please remove it

Reply:
Oops, forgot to remove it